engines/pisa-0.8.1/tools/build_index.cpp

```diff
@@ -1,12 +1,66 @@
 #include <spdlog/sinks/stdout_color_sinks.h>
 #include <spdlog/spdlog.h>
 
+#include <Porter2.hpp>
+#include <boost/algorithm/string.hpp>
+#include <fmt/format.h>
+#include <forward_index_builder.hpp>
+#include <invert.hpp>
+#include <mio/mmap.hpp>
 #include <nlohmann/json.hpp>
+#include <tbb/task_scheduler_init.h>
 
-int main(int argc, char const *argv[])
+static std::size_t const THREADS = 4;
+static std::size_t const BATCH_SIZE = 10'000;
+static std::string const FWD = "/tmp/fwd";
+static std::string const INV = "/tmp/inv";
+
+using pisa::Document_Record;
+using pisa::Forward_Index_Builder;
+
+void parse()
+{
+    pisa::Forward_Index_Builder fwd_builder;
+    fwd_builder.build(
+        std::cin,
+        FWD,
+        [](std::istream& in) -> std::optional<Document_Record> {
+            std::string line;
+            if (std::getline(in, line) && not line.empty()) {
+                auto record = nlohmann::basic_json<>::parse(line);
+                auto url = record["url"].get<std::string>();
```

Comment:
The format of the corpus is:
```
{"id": "https://en.wikipedia.org/wiki?curid=48687903", "text": " jeon hye ... on july "}
```

I propose to use 
```
#include <boost/regex.hpp>
...
boost::smatch what;
boost::regex_match(response, record["url"].get<std::string>(), "https://en.wikipedia.org/wiki?curid=([0-9]+)");
```

Reply:
I think this one is OK if we use the `corpus_transform.py` beforehand

Reply:
I don't know why I thought it was different, I might have been looking at the wrong file.

Reply:
are you sure about this? I have a file on `moa` that has this format:
```json
{"url":"https://en.wikipedia.org/wiki?curid=48687903","title":"Jeon Hye-jin (actress, born 1988)","body":"\nJeon Hye-jin (actress, born 1988)\n\nJeon Hye-jin (born June 17, 1988) is a South Korean actress.\nPersonal life.\nJeon married his \"Smile, You\" co-star Lee Chun-hee on March 11, 2011. Their daughter, Lee So Yu, was born on July 30, 2011.\n\n"}
```

Reply:
Confirmed, I downloaded the collection linked on github, unzipped and this is what I get:

```
{"url":"https://en.wikipedia.org/wiki?curid=48687903","title":"Jeon Hye-jin (actress, born 1988)","body":"\nJeon Hye-jin (actress, born 1988)\n\nJeon Hye-jin (born June 17, 1988) is a South Korean actress.\nPersonal life.\nJeon married his \"Smile, You\" co-star Lee Chun-hee on March 11, 2011. Their daughter, Lee So Yu, was born on July 30, 2011.\n\n"}
{"url":"https://en.wikipedia.org/wiki?curid=48687919","title":"Benham, Indiana","body":"\nBenham, Indiana\n\nBenham is an unincorporated community in Ripley County, in the U.S. state of Indiana.\nHistory.\nAn old variant name of the community was Benhams Store. A post office opened under the name Benham Store in 1866, the name was shortened to Benham 1888, and the post office was discontinued in 1934. John Benham, Jr., served as a first postmaster.\n\n"}
```